File: SPADE/Agente.py
import ast
import spade
from spade.message import Message
from spade.behaviour import FSMBehaviour, State

INICIO = "INICIO"
CONTROL = "CONTROL"

class AgentBehaviour(FSMBehaviour):
    async def on_start(self):
        print(f"{self.agent.name} >> INIT")

    async def on_end(self):
        print(f"{self.agent.name} >> FIN")
        await self.agent.stop()


class inicio(State):
    async def run(self):
        msg = Message(to="tournamentManager@localhost")
        msg.body = ""  

        await self.send(msg)
        print(f"{self.agent.name} >> Notificacion enviada")
        self.set_next_state(CONTROL)


class control(State):
    async def run(self):
        msg = await self.receive(timeout=1)
        if msg:
            print(f">> {msg.body}")
            if(msg.body == "STOP"):
                await self.agent.stop()
                return
            res = ast.literal_eval(msg.body)
            msg = Message(to="tournamentManager@localhost")
            msg.body = str(self.agent.strategy(self, res[0], res[1]))
            await self.send(msg)

        self.set_next_state(CONTROL)


class Agent(spade.agent.Agent):
    # default constructor
    def __init__(self, strategy, *args):
        self.strategy = strategy
        super().__init__(*args)

    async def setup(self):
        # Definir la Maquina de Estados Finitos
        fsm = AgentBehaviour()

        # Estados
        fsm.add_state(name=INICIO, state=inicio(), initial=True)
        fsm.add_state(name=CONTROL, state=control())

        # Transiciones entre estados
        fsm.add_transition(source=INICIO, dest=CONTROL)
        fsm.add_transition(source=CONTROL, dest=CONTROL)

        self.add_behaviour(fsm)

File: JADE/Agente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package torneo;

import jade.core.AID;
import jade.core.Agent;
import jade.core.behaviours.FSMBehaviour;
import jade.core.behaviours.OneShotBehaviour;
import jade.lang.acl.ACLMessage;
import java.io.FileWriter;
import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author Ximo
 */
public class Agente extends Agent {

    Strategy s;
    private FileWriter fw;
    private final String INICIO = "INICIO",
            CONTROL = "CONTROL",
            FIN = "FIN";

    public class Strategy {

        public int calc(int roundNum, int last) {
            return 0;
        }
    }

    public void setup() {
        try {
            fw = new FileWriter(getLocalName() + ".txt");
        } catch (IOException e) {
            System.out.println(e);
        }
        print("STARTED");

        // Definir la Maquina de Estados Finitos
        FSMBehaviour fsm = new FSMBehaviour(this) {
            public int onEnd() {
                try {
                    fw.close();
                } catch (IOException ex) {
                    Logger.getLogger(TournamentManager.class.getName()).log(Level.SEVERE, null, ex);
                }
                doDelete();
                return super.onEnd();
            }
        };

        // Registrar los estados
        fsm.registerFirstState(new Inicio(), INICIO);
        fsm.registerState(new Control(), CONTROL);
        fsm.registerLastState(new Fin(), FIN);

        // Transiciones entre estados
        fsm.registerDefaultTransition(INICIO, CONTROL);
        fsm.registerTransition(CONTROL, CONTROL, 0);
        fsm.registerTransition(CONTROL, FIN, 1);

        addBehaviour(fsm);
    }

    private class Inicio extends OneShotBehaviour {

        public void action() {
            AID tm = new AID("TournamentManager", AID.ISLOCALNAME);
            ACLMessage msg = new ACLMessage(ACLMessage.INFORM);
            msg.addReceiver(tm);
            msg.setContent("");
            send(msg);
        }
    }

    private class Control extends OneShotBehaviour {
        int nextState = 0;
        ACLMessage msg;

        public void action() {
            msg = blockingReceive();
            print(">> " + msg.getContent());
            if (msg.getContent().equals("STOP")) {
                nextState = 1;
                return;
            }
            String[] args = msg.getContent().split(" ");
            AID tm = msg.getSender();
            msg = new ACLMessage(ACLMessage.INFORM);
            msg.addReceiver(tm);
            msg.setContent("" + s.calc(Integer.parseInt(args[0]), Integer.parseInt(args[1])));
            send(msg);
        }
        public int onEnd() {
            return nextState;
        }
    }

    private class Fin extends OneShotBehaviour {

        public void action() {
            print("Terminado.");
        }
    }

    private void print(String s) {
        try {
            fw.write(s + "\n");
        } catch (IOException ex) {
            Logger.getLogger(TournamentManager.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}
